File: csharp/HearthstoneReplays/Parser/ReplayData/GameActions/TagChange.cs
#region

using System.Xml.Serialization;

#endregion

namespace HearthstoneReplays.Parser.ReplayData.GameActions
{
	public class TagChange : GameAction
	{
		[XmlAttribute("tag")]
		public int Name { get; set; }

		[XmlAttribute("value")]
		public int Value { get; set; }

		public override bool Equals(object obj)
		{
			var other = obj as TagChange;
			if(other == null)
				return false;
			return other.Name == Name && other.Value == Value;
		}
	}
}

File: java/src/main/java/com/hearthsim/hsreplay/parser/replaydata/gameactions/TagChange.java
package com.hearthsim.hsreplay.parser.replaydata.gameactions;

import javax.xml.bind.annotation.XmlAccessOrder;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorOrder;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@EqualsAndHashCode(of = { "name", "value" }, callSuper = false)
@NoArgsConstructor
@XmlAccessorType(XmlAccessType.FIELD)
@XmlAccessorOrder(XmlAccessOrder.ALPHABETICAL)
@ToString
public class TagChange extends GameAction {

	@XmlAttribute(name = "tag")
	private int name;

	@XmlAttribute(name = "value")
	private int value;

	public TagChange(int entity, int name, int value) {
		super(entity);
		this.name = name;
		this.value = value;
	}

}
